src/protocol/resp/src/request/mod.rs

```diff
@@ -86,9 +86,10 @@ impl Parse<Request> for RequestParser {
                 }
             }
 
-            if &remaining[0..2] != b"\r\n" {
-                return Err(Error::from(ErrorKind::WouldBlock));
-            }
+            match &remaining.get(0..2) {
+                Some(slice) if slice != b"\r\n" => return Err(Error::from(ErrorKind::WouldBlock)),
+                _ => (),
+            };
```

Comment:
You can probably use [`starts_with`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.starts_with) here. It would be a little cleaner :)

Reply:
ty @swlynch99

Reply:
@swlynch99 actually I think this might introduce some disrepancies in certain cases if we deviate from the direct equality checking.

With the if-conditional-equality we get this behavior:
```
> echo -e -n 'get test\r\nget test\r\n' | nc localhost 12321
$-1
$-1
```
when a client sends commands packed together like shown.

With the `starts_with` we get this behavior:
```
❯ echo -e -n 'get test\r\nget test\r\n' | nc localhost 12321
```
with no response coming back. (Not sure if it would trigger a response w/ some additional prompting but I don't know how to both send a very specific set of bytes with netcat *and* then make it interactive and don't feel like digging further)

@brayniac I'm going to assume we want to keep the `slice != b"\r\n"` based on the findings above

Reply:
I actually meant something like this
```suggestion
            if !remaining.starts_with(b"\r\n") {
                return Err(Error::from(ErrorKind::WouldBlock));
            }
```
which should have the same behaviour as the whole match statement.

Reply:
@swlynch99 ok cool thanks for the clarification

Reply:
I like @swlynch99's suggestion here

Reply:
@brayniac yeah I had misinterpreted what he was telling me to do, after he clarified it worked as expected